### PR TITLE
Support object representation of `StudyDirection` for `create_study` arguments

### PR DESCRIPTION
**Pull request by g-votte**
This PR enables to pass `StudyDirection` objects to the `direction`/`directions` argument of `optuna.create_study`.

## Motivation
Supporting `StudyDirection` objects sometimes simplifies the user’s code especially when the user already has the `StudyDirection` objects when creating the study, e.g., copying an existing study to new one as follows.

```python
study_src = optuna.load_study(study_name=study_name, storage=storage_url)
study_dist = optuna.create_study(directions=study_src.directions)

for t in study_src.trials:
    study_dist.add_trial(t)
```
Note that the code above raises an error in the second line without applying this PR, and the user has to convert the direction objects to the string representation.

### CODE REVIEW

File: optuna/study.py [hunk 1]
```diff
@@ -1104,12 +1105,21 @@ def objective(trial):
...
-        StudyDirection.MINIMIZE if d == "minimize" else StudyDirection.MAXIMIZE for d in directions
-    ]
+    direction_objects = []
+    for d in directions:
+        if d == "minimize":
+            direction_objects.append(StudyDirection.MINIMIZE)
+        elif d == "maximize":
+            direction_objects.append(StudyDirection.MAXIMIZE)
+        else:
+            assert isinstance(d, StudyDirection)
```
Comment (g-votte) [inline optuna/study.py:1121]: I've added this assertion to pass the mypy check. Please let me know if there's much smarter way.

Comment (hvy) [reply optuna/study.py:1121]: Not sure this addresses your comment but we could also write something like this.
```python
direction_objects = [d if isintance(d, StudyDirection) else StudyDirection[d.upper()] for d in directions]
```

**Review by hvy — APPROVED** [review]
Thanks, LGTM! Left a minor comment but it's mostly a cosmetic.

File: optuna/study.py (hunk 1 above)
Comment (g-votte) [reply optuna/study.py]: Cool! I think this is much simpler and resolves the redundant assertion.

File: optuna/study.py [hunk 2]
```diff
@@ -1104,11 +1105,14 @@ def objective(trial):
 
     if len(directions) < 1:
         raise ValueError("The number of objectives must be greater than 0.")
-    elif any(d != "minimize" and d != "maximize" for d in directions):
+    elif any(
+        d not in ["minimize", "maximize", StudyDirection.MINIMIZE, StudyDirection.MAXIMIZE]
+        for d in directions
+    ):
         raise ValueError("Please set either 'minimize' or 'maximize' to direction.")
```
Comment (toshihikoyanase) [inline optuna/study.py:1112]: A nit, but how about adding `StudyDirection` to the error message?
```suggestion
    ):
        raise ValueError(
            "Please set one of 'minimize', 'maximize' or a StudyDirection member to direction."
        )
```

**Review by toshihikoyanase — COMMENTED** [review]
Thank you for your PR. This feature definitely improves the usability of `optuna.create_study`.
The change looks good to me, but I have a small comment on the error message.

File: optuna/study.py (hunk 2 above)
Comment (g-votte) [reply optuna/study.py]: Thanks for your suggestion! I updated the error message.

Comment (toshihikoyanase) [reply optuna/study.py]: Thank you for your update. It's better than my proposal 👍

**Review by toshihikoyanase — APPROVED** [review]
Thank you for your update. LGTM!

**Pull request merged**